### PR TITLE
Move invalid files to 'rejected' container

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-679

### Change description ###

Move invalid files to separate, '...-rejected' containers.
Also, introduce FILE_VALIDATION_FAILED event type, as now we distinguish between general document processing failure and validation failure.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lgonczar** [comment]
Actually, I'll split it into two commits - the second one introducing the new event type

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 1]
```diff
@@ -186,8 +189,13 @@ private ZipFileProcessor processZipFileContent(
             zipFileProcessor.setEnvelope(envelopeProcessor.saveEnvelope(envelope));
 
             processor = zipFileProcessor;
+        } catch (InvalidEnvelopeSchemaException
+            | FileNameIrregularitiesException
+            | NonPdfFileFoundException
+            | MetadataNotFoundException ex) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: CI: `)` newline is it not?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 2]
```diff
@@ -201,4 +209,12 @@ private boolean isReadyToBeProcessed(CloudBlockBlob blob) {
         java.util.Date cutoff = Date.from(Instant.now().minus(this.blobProcessingDelayInMinutes, ChronoUnit.MINUTES));
         return blob.getProperties().getLastModified().before(cutoff);
     }
+
+    private void handleInvalidFileError(
+        Event fileValidationFailure, String containerName, String zipFilename,
+        Exception cause) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: `)` defo new line ;). oh, and all the arguments

**Comment by lgonczar** [comment]
Closing this pull request, as master is currently unavailable.

**Pull request merged**